### PR TITLE
feat(argo-workflows): add values, secret and configmap for static token auth

### PR DESCRIPTION
**Pull request by LoricAndre**
Signed-off-by: Loric ANDRE <loric.a@padoa-group.com>

Linked to https://github.com/argoproj/argo-workflows/pull/9086

This requires that PR to be merged to be fully fonctional as this implements the k8s part of a new authentication mode for argo-workflow (--auth-mode token using a static token from a kube secret).

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-workflows/Chart.yaml
Comment (jmeridth) [inline charts/argo-workflows/Chart.yaml:6]: do you plan to wait for 3.3.9 or do you plan to keep this TODO in here?  I [queried argo-workflows maintainers](https://cloud-native.slack.com/archives/C020XM04CUW/p1656525154019479?thread_ts=1656525051.021529&cid=C020XM04CUW) and they said 3.3.9 release would be next month sometime, may be a bit.

**Review by jmeridth — COMMENTED** [review]
Looks good but one question.  I'm going to approve the test run but not the PR yet.

File: charts/argo-workflows/Chart.yaml
Comment (jmeridth) [reply charts/argo-workflows/Chart.yaml:6]: May want to put this into draft mode then, wdyt?

Comment (jmeridth) [reply charts/argo-workflows/Chart.yaml:6]: Also, that kind of comment would be great in the PR description.

Comment (LoricAndre) [reply charts/argo-workflows/Chart.yaml:6]: I made sure this does not change anything by default outside of the version of the doc, but I'll switch it to a draft, you're right.

Comment (jmeridth) [reply charts/argo-workflows/Chart.yaml:6]: I missed the linked PR.  You're good. 😄

File: charts/argo-workflows/values.yaml
Comment (mkilchhofer) [inline charts/argo-workflows/values.yaml:490]:
```suggestion
  ## secret for token authentication
  authTokenSecret:
    # -- Enable secret for token authentication
    # this is required if using `--auth-mode sso` to the server command line.
    enabled: false
    # -- Auto create secret
    create: true
    # -- Default secret name
    name: '{{ include "argo-workflows.fullname" . }}-static-token'
    ## Set token from value (unsafe)
    # token: test-token
```

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
I'd prefer to have all "leaf" YAML keys documented via helm-docs instead of one big object

File: charts/argo-workflows/Chart.yaml
Comment (jmeridth) [inline charts/argo-workflows/Chart.yaml:6]: #1393 Once it merges I'll comment here.  CHeers.

**Comment by mkilchhofer** [comment]
Awaiting upstream PR was closed by the stale bot: https://github.com/argoproj/argo-workflows/pull/9086

Closing this one here also. @LoricAndre feel free to reopen this PR or file a new one if you got this feature into argo-workflows.

**Pull request closed without merging**
